bigtable/client/instance_admin.cc

```diff
@@ -107,6 +107,77 @@ google::bigtable::admin::v2::Instance InstanceAdmin::CreateInstanceImpl(
   return result;
 }
 
+std::future<google::bigtable::admin::v2::Instance>
+InstanceAdmin::UpdateInstance(google::bigtable::admin::v2::Instance* instance,
+                              google::protobuf::FieldMask* update_mask) {
+  return std::async(std::launch::async, &InstanceAdmin::UpdateInstanceImpl,
+                    this, instance, update_mask);
+}
+
+google::bigtable::admin::v2::Instance InstanceAdmin::UpdateInstanceImpl(
+    google::bigtable::admin::v2::Instance* instance,
+    google::protobuf::FieldMask* update_mask) {
+  // Copy the policies in effect for the operation.
+  auto rpc_policy = impl_.rpc_retry_policy_->clone();
+  auto backoff_policy = impl_.rpc_backoff_policy_->clone();
+  MetadataUpdatePolicy metadata_update_policy(instance->name(),
+                                              MetadataParamTypes::NAME);
+
+  google::bigtable::admin::v2::PartialUpdateInstanceRequest request;
+  request.mutable_instance()->Swap(instance);
+  request.mutable_update_mask()->Swap(update_mask);
+
+  using ClientUtils =
+      bigtable::internal::noex::UnaryClientUtils<InstanceAdminClient>;
+
+  grpc::Status status;
+  auto response = ClientUtils::MakeCall(
+      *impl_.client_, *rpc_policy, *backoff_policy,
+      impl_.metadata_update_policy_, &InstanceAdminClient::UpdateInstance,
+      std::move(request), "InstanceAdmin::UpdateInstance", status, false);
+  if (not status.ok()) {
+    bigtable::internal::RaiseRpcError(status,
+                                      "unrecoverable error in MakeCall()");
+  }
+
+  google::bigtable::admin::v2::Instance result;
+  do {
+    if (response.done()) {
+      if (response.has_response()) {
+        auto const& any = response.response();
+        if (not any.Is<google::bigtable::admin::v2::Instance>()) {
+          google::cloud::internal::RaiseRuntimeError("invalid result type");
+        }
+        any.UnpackTo(&result);
+        return result;
+      }
+      if (response.has_error()) {
+        bigtable::internal::RaiseRpcError(
+            grpc::Status(static_cast<grpc::StatusCode>(response.error().code()),
+                         response.error().message()),
+            "long running op failed");
+      }
+    }
+    // Wait before polling, and then poll the operation to get the new
+    // "response.
+    auto delay = backoff_policy->on_completion(status);
```

Comment:
Please add a `TODO()` here to use the PollingPolicy once #461 is merged.

Reply:
Should I create an issue and mention it with TODO?

Reply:
That sounds good.  Make sure the new issue is included in the same milestone.

Reply:
#578 created for this purpose.